### PR TITLE
Updated the theme with types

### PR DESCRIPTION
**Pull request by zecuria**
Added types to theme

### CODE REVIEW

File: src/themes/default/index.ts [hunk 1]
```diff
@@ -0,0 +1,318 @@
+import accordion from "./accordion";
```
Comment (alifa20) [inline src/themes/default/index.ts:1]: Isn't this index file too big ? Maybe we can refactor into smaller files ?

File: src/themes/default/list.ts [hunk 2]
```diff
@@ -1,37 +1,42 @@
...
+import { BasisTheme, ListTheme, ThemeHelpers } from "../types";
+
+export default (
+  theme: Pick<BasisTheme, "colors" | "space" | "radii" | "fontWeights">,
+  { getTextStyle }: ThemeHelpers
+): ListTheme => {
   return {
-    getCSS: ({ targetElement, type, variant, textStyle }) => {
-      switch (targetElement) {
+    getCSS: (options) => {
```
Comment (alifa20) [inline src/themes/default/list.ts:8]: Destructuring the param seems to be a better and nicer code

File: src/themes/default/text.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { TextTheme, ThemeHelpers } from "../types";
+
+export default (_theme: {}, { getColor }: ThemeHelpers): TextTheme => {
+  return {
+    getCSS: ({ color, align, wrap }) => {
+      return {
```
Comment (alifa20) [inline src/themes/default/text.ts:6]: remove return ? same as what you have in `src/themes/default/textStyles.ts`
https://github.com/moroshko/basis/pull/247/files#diff-5f026162d660728ff22e62a54e9e9e1046a66effbdb96cc9abe98dba2d9a6b9fR9

File: src/themes/default/list.ts (hunk 2 above)
Comment (zecuria) [reply src/themes/default/list.ts:8]: Can't destructured it, some of the properties are only available to some `targetElement`. see: https://github.com/moroshko/basis/pull/247/files#diff-9bdb7564c7c64bf70f00b13cc604c7e53f8994437029037ea3ae38671fabfc9fR236

As such typescript would give errors as not all target elements have all the properties

File: src/themes/default/index.ts (hunk 1 above)
Comment (zecuria) [reply src/themes/default/index.ts:1]: It was like this from before. This PR isn't addressing any refactor, it's simply adding typescript support

File: src/themes/default/text.ts (hunk 3 above)
Comment (zecuria) [reply src/themes/default/text.ts:6]: It was like this before the change, see: https://github.com/moroshko/basis/pull/247/files/150f7c89fd2264570d1a9e2c6ada2cf2fc762cdd#diff-00a4e15ca1cf2270a4bfa851239d4d20ee055a94d0647c74a5abc84c6a431c39L4

File: src/hooks/useTheme.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import defaultTheme from "../themes/default";
+import { EnhancedTheme } from "../themes/types";
+import { enhanceTheme } from "../utils/theme";
+
+export const ThemeContext = React.createContext<EnhancedTheme>(
+  enhanceTheme(defaultTheme)
+);
+
+function useTheme() {
+  const theme = useContext(ThemeContext);
```
Comment (strahinjaajvaz) [inline src/hooks/useTheme.ts:11]: can't we just return the `useContext(ThemeContext)` without creating the theme variable?

**Review by strahinjaajvaz — APPROVED** [review]

File: src/themes/types.ts [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+    heading: string;
+    body: string;
+  };
+  fontWeights: {
+    light: number;
+    medium: number;
+    semiBold: number;
+    bold: number;
+  };
+  colors: {
```
Comment (moroshko) [inline src/themes/types.ts:405]: Can't we use the `type ThemeColors` defined above?

File: src/themes/types.ts [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+  checkbox: CheckboxTheme;
+  accordion: AccordionTheme;
+};
+
+export type EnhancedTheme = BasisTheme & {
+  minMediaQueries: Record<Breakpoint, string>;
+  exclusiveMediaQueries: Record<Breakpoint, string>;
+  getColor: (color: Color) => string | null;
+  getTextStyleCSS: (textStyle: TextStyleNames) => CSSObject | null;
+  getSpaceValue: (spcae: string | number) => string | null;
```
Comment (moroshko) [inline src/themes/types.ts:553]: Typo: spcae

**Review by moroshko — APPROVED** [review]
Awesome work!
Just minor comments, and please check in Percy that there is no visual regression before merging.

**Pull request merged**